internal/model/function.go

```diff
@@ -0,0 +1,114 @@
+package model
+
+import (
+	"github.com/dave/dst"
+)
+
+// Function represents a method declared on a struct type.
+type Function struct {
+	Name              string        // Name of the function/method
+	Receiver          TypeReference // Type of the receiver (pointer or value)
+	IsPointerReceiver bool          // Whether the receiver is a pointer
+	Parameters        []Parameter   // Function parameters
+	Results           []Parameter   // Return values
+	description       []string
+	declaredOn        *Object // The object this function is declared on
+}
+
+// Parameter represents a function parameter or return value.
+type Parameter struct {
+	Name string        // Name of the parameter (may be empty)
+	Type TypeReference // Type of the parameter
+}
```

Comment:
Prefer to keep each significant type in a separate file to make discovery easier for future maintainers. Move `Parameter` and any related methods into a separate file.